pkg/clustermesh/types/types.go

```diff
@@ -4,18 +4,33 @@
 package types
 
 import (
-	"errors"
 	"fmt"
+
+	"github.com/cilium/cilium/pkg/defaults"
 )
 
 const (
 	// ClusterIDMin is the minimum value of the cluster ID
-	ClusterIDMin = 0
-
-	// ClusterIDMax is the maximum value of the cluster ID
-	ClusterIDMax = 255
+	ClusterIDMin    = 0
+	ClusterIDExt511 = 511
 )
 
+// ClusterIDMax is the maximum value of the cluster ID
+var ClusterIDMax uint32 = defaults.MaxConnectedClusters
```

Comment:
Not sure how feasible it is in this case without an extensive refactoring, but we should strive to avoid using run-time initialized global variables as much as possible. They tend to be problematic especially in combination with the hive framework, as depending on the initialization order of the cells they might, or might not be already correctly initialized (I see that you are initializing this before starting the hive, which is safe, but there's still an ongoing effort to move the registration of the flags to the appropriate places, which might change it).

In this specific case, I'd suggest wrapping this in an object which can be used to register the flags (so that it conforms with he Hive approach, and it can also be reused for the clustermesh-apiserver) and perform the validation. This way, the components which are already cells (e.g., clustermesh) can depend on it, while the others can still rely on the global variable to prevent an extensive refactoring. The same object could also be used to do the validation of the `ClusterConfig`, relieving the main ClusterMesh object from that task,

Reply:
ah yes, I like this idea. I already began a refactor to remove this global, but I had planned on replacing it just with usage of the DaemonConfig field. I like your idea better

Reply:
in lieu of removing the `ClusterIDMax` global entirely (I began this several times, but the refactor ballooned. Probably best for its own PR), I've moved maxConnectedClusters into `ClusterInfo` , which validates and populates `ClusterIDMax` to be used in places that haven't been modularized yet.